2016-06-28-Rust-Reach-Fractal.md

```diff
@@ -0,0 +1,91 @@
+
+## A path forward: Rust, Reach, and Fractal
+
+Drawing on some exploratory work we did over a year ago, we think we can work on these problems such that their solutions build upon each other. Here's the idea:
+
+1. Work on improved **Rust** support. The Rust language executes code much faster and more robustly than JavaScript and should serve well in production. The code is overall pretty legible, and it's a welcoming community with good documentation. We've already begun tooling for this on the [tessel-rust](//github.com/tessel/tessel-rust) repo. Tessel Project work on Rust doesn't mean reduced support for or interest in JavaScript – we think they have the potential to play well together and want both to have first-class support.
+1. Build on this for our **Tessel Reach** project: cheap, low-power boards which break out to Tessel module ports. Think of a Reach board as an extension cable for a module– this is how it should feel to use one. Except instead of the long wires and limited number of modules, Tessel Reach boards will connect wirelessly to a USB BLE dongle on T2 and not be limited by the number of module ports available on the Tessel 2. You can see a bit more detail on [this issue](https://github.com/tessel/project/issues/142).
+1. Use the JavaScript --> Rust compilation work to begin building the **Fractal** project: software tooling to help you optimize your prototype (software and hardware) into production. Explicit hardware requirements and memory-efficient software can let us begin to suggest hardware optimization. Eventually, this would move toward easily comprehensible messaging, such as "compile to Rust and reduce your file size of X to Y", "you aren't using Module Port B, remove it?" and "you're only using Y memory- you may want to consider this cheaper chip (link)".
+
+## What do you think?
+
+This is the beginning of a plan. It's an outline of a roadmap of a vision, and it's going to be a lot of work.
+
+For me, it's an inspiration. I want to work in the context of a grand plan. And it's my hope that this grand plan excites and inspires you, too.
+
+When it comes down to it, that's the crux. What do you think? Is this something you're excited about building, and do these tools feel like they would be useful to you? Per our final core philosophy:
+
+> **Community matters.** This project should be a welcoming, inclusive, and respectful place. It’s important to communicate, to question, and to come together.
+
+Please reach out on -ISSUE ON TESSEL/PROJECT?- and let us know what you think. We need to hear if you're excited, if you're unexcited, or if there's something in particular you want to do. Because in the end, the Tessel Project is built out of the community, and we can't do it without you.
```

Comment:
I think a "Vision" issue is fine but it should have a definite lifetime (ie we'll close this from comments on X date) and it should be made clear that we're not looking for consensus.